core/src/main/scala/org/apache/spark/api/r/SerDe.scala

```diff
@@ -355,6 +355,13 @@ private[spark] object SerDe {
           writeInt(dos, v.length)
           v.foreach(elem => writeObject(dos, elem))
 
+        // Handle Properties
```

Comment:
my preference is to do more in R. if you feel strongly about having a helper in Scala instead of handling Properties then we could move most of the code into a Scala helper.